crates/fj-core/src/objects/object_set.rs

```diff
@@ -28,16 +28,17 @@ impl<T> ObjectSet<T> {
     /// # Panics
     ///
     /// Panics, if the iterator contains duplicate `Handle`s.
-    pub fn new(handles: impl IntoIterator<Item = Handle<T>>) -> Self
+    pub fn new<H>(handles: impl IntoIterator<Item = H>) -> Self
     where
         T: Debug + Ord,
+        H: Into<HandleWrapper<T>>,
```

Comment:
This can be written a bit more compactly:

```suggestion
    pub fn new(handles: impl IntoIterator<Item = impl Into<HandleWrapper<T>>) -> Self
    where
        T: Debug + Ord,
```

This is the style I usually prefer, but it's not actually that much better, in any meaningful way. I'll merge this anyway, but I figured I'd mention the possibility, in case you weren't aware.